core/include/jiminy/core/robot/AbstractTransmission.h

```diff
@@ -0,0 +1,297 @@
+        /// \details    It is the transformation matrix to convert quantities from motor to joint-level.
+        ///////////////////////////////////////////////////////////////////////////////////////////////
+        matrixN_t const & getForwardTransform(void) const;
+
+        ///////////////////////////////////////////////////////////////////////////////////////////////
+        /// \brief      Get backwardTransform_.
+        ///
+        /// \details    It is the transformation matrix to convert quantities from joint to motor-level.
+        ///////////////////////////////////////////////////////////////////////////////////////////////
+        matrixN_t const & getInverseTransform(void) const;
+
+        ///////////////////////////////////////////////////////////////////////////////////////////////
+        /// \brief      Compute forward transmission.
+        ///
+        /// \details    Compute forward transmission from motor to joint.
+        ///////////////////////////////////////////////////////////////////////////////////////////////
+        hresult_t computeForward(float64_t const & t,
+                                 vectorN_t & q,
+                                 vectorN_t & v,
+                                 vectorN_t & a,
+                                 vectorN_t & uJoint);
+
+        ///////////////////////////////////////////////////////////////////////////////////////////////
+        /// \brief      Compute backward transmission.
+        ///
+        /// \details    Compute backward transmission from joint to motor.
+        ///////////////////////////////////////////////////////////////////////////////////////////////
+        hresult_t computeBackward(float64_t const & t,
+                                  vectorN_t const & q,
+                                  vectorN_t const & v,
+                                  vectorN_t const & a,
+                                  vectorN_t const & uJoint);
+
+    protected:
+        ///////////////////////////////////////////////////////////////////////////////////////////////
+        /// \brief      Request the transmission to update its actual state based of the input data.
+        ///
+        /// \details    It assumes that the internal state of the robot is consistent with the
+        ///             input arguments. This transformation is used to compute the forward transformation
+        ///             of the transmission transforming motor position, velocity and torque to joint
+        ///             position, velocity and torque.
+        ///
+        /// \param[in]  q        Current configuration of the motors.
+        /// \param[in]  v        Current velocity of the motors.
+        /// \param[out] out      Matrix transforming quantities from motor to joint level
+        ///////////////////////////////////////////////////////////////////////////////////////////////
+        virtual void computeTransform(Eigen::VectorBlock<vectorN_t const> const & q,
```

Comment:
Remove `v` dependency. We will stick to holonomic transmission. It is much simpler and cover 99% of real applications.